### ISSUE TITLE
Problem with building

### ISSUE THREAD

**Issue by ge0rdi** [CircuitMess/Bit-Firmware#71]
Hello.

I was trying to build Bit firmware but had some troubles with it. Building instructions say:

> The production firmware is built using IDF version 5.1, commit [3a45d4e](https://github.com/espressif/esp-idf/tree/3a45d4e949a174e8829a2e4c86c421b030ceac5a).

Using version from mentioned commit results in many build errors.
It seems that latest firmware was built using IDF `5.1.1` (it can be seen in firmware binary).
With that version it builds (almost) fine.

Then there is a problem with `LovyanGFX` submodule. Version `1.1.8` doesn't work well with IDF 5.1.1 so at least `1.1.12` has to be used (I've tried `1.1.16` that seems to work just fine).

Could you, please, fix build instructions to point to proper IDF version and also update `LovyanGFX` to more recent version?

**Comment by JustZvan** [comment]
Hey! I'm also having issues with building. I tried your steps, by using the latest master branch of LovyanGFX, still doesn't work. This time it isn't the libraries and it's the source code itself. Could you share some steps on how do i actually build it?

**Comment by ge0rdi** [comment]
> using the latest master branch of LovyanGFX, still doesn't work

I'm not sure if it works with latest master.
I have used `1.1.16` tag.

Here are instructions to build my modified firmware (along with micro python):
https://gist.github.com/ge0rdi/7e9aa11967b6f70f7338ca6be137adee

It works for me that way.

**Comment by MiLeG** [comment]
Hi @ge0rdi !

The [README ](https://github.com/CircuitMess/Bit-Firmware/blob/master/README.md)is now updated and should contain all the necessary steps for building.

Let us know if this works for you.

**Comment by MiLeG** [comment]
Ok, the patch file had some mangled encoding.
Should be fixed now with 0eed7d6.

@ge0rdi Bumping the LovyanGFX version to v1.1.16 is a valid solution.
However, during development there was some swapping between different ESP-IDF versions and the LovyanGFX library wasn't functional at the time with our selected ESP-IDF version and required some patching.

Since LovyanGFX is being (mostly) developed by one person, there's always the risk of some unexpected bug or regression being introduced in new releases. We stuck with the v1.1.8 release since we used it previously and didn't have much issues (aside from the single patched line).

In short, it was just a preferential choice with what was available at the time of development.

**Comment by ge0rdi** [comment]
Build now works without a problem.
Thank you for the fix.
